openrtb3/openrtb3.go

```diff
@@ -0,0 +1,162 @@
+	Deal   string          `json:"deal,omitempty"`   // Reference to a deal from the bid request if this bid pertains to a private marketplace deal; specifically “deal.id”.
+	CID    string          `json:"cid,omitempty"`    // Campaign ID or other similar grouping of brand-related ads. Typically used to increase the efficiency of audit processes.
+	Tactic string          `json:"tactic,omitempty"` // Tactic ID to enable buyers to label bids for reporting to the exchange the tactic through which their bid was submitted. The specific usage and meaning of the tactic ID should be communicated between buyer and exchanges a priori.
+	PURL   string          `json:"purl,omitempty"`   // Pending notice URL called by the exchange when a bid has been declared the winner within the scope of an OpenRTB compliant supply chain (i.e., there may still be non-compliant decisioning such as header bidding). Substitution macros may be included.
+	BURL   string          `json:"burl,omitempty"`   // Recommended. Billing notice URL called by the exchange when a winning bid becomes billable based on exchange-specific business policy (e.g., markup rendered). Substitution macros may be included.
+	LURL   string          `json:"lurl,omitempty"`   // Loss notice URL called by the exchange when a bid is known to have been lost. Substitution macros may be included. Exchange-specific policy may preclude support for loss notices or the disclosure of winning clearing prices resulting in ${OPENRTB_PRICE} macros being removed (i.e., replaced with a zero-length string).
+	Exp    int64           `json:"exp,omitempty"`    // Advisory as to the number of seconds the buyer is willing to wait between auction and fulfilment.
+	MID    string          `json:"mid,omitempty"`    // ID to enable media to be specified by reference if previously uploaded to the exchange rather than including it by value in the domain objects.
+	Macro  []Macro         `json:"macro,omitempty"`  // Array of “Macro” objects that enable bid specific values to be substituted into markup; especially useful for previously uploaded media referenced via the “mid” attribute. Refer to Object: Macro.
+	Media  json.RawMessage `json:"media,omitempty"`  // Layer-4 domain object structure that specifies the media to be presented if the bid is won conforming to the specification and version referenced in “openrtb.domainspec” and “openrtb.domainver”. For AdCOM v1.x, the objects allowed here are “Ad” and any objects subordinate thereto as specified by AdCOM.
+	Ext    json.RawMessage `json:"ext,omitempty"`    // Optional demand source specific extensions.
+}
+
+// Macro object constitutes a buyer defined key/value pair used to inject dynamic values into media markup.
+// While they apply to any media markup irrespective of how it is conveyed, the principle use case is for media that was uploaded to the exchange prior to the transaction (e.g., pre-registered for creative quality review) and referenced in bid.
+type Macro struct {
```

Comment:
BLOCKED/WARNING - implementation may be wrong: https://github.com/InteractiveAdvertisingBureau/openrtb/issues/10

Reply:
Clarified, this was the right approach.